File: src/UnwindMC.Tests/Util/MiscExtensions.cs
using System;
using System.Linq;

namespace UnwindMC.Tests.Helpers
{
    public static class MiscExtensions
    {
        public static string StripIndent(this string str)
        {
            var lines = str.Replace("\r\n", "\n").Split('\n');
            int indent = lines
                .Skip(1)
                .Min(l => l.TakeWhile(c => c == ' ').Count());
            return lines[0] + Environment.NewLine +
                string.Join(Environment.NewLine, lines
                    .Skip(1)
                    .Select(l => l.Substring(indent)));
        }
    }
}

File: src/UnwindMC.Tests/Util/MiscExtensions.fs
[<AutoOpen>]
module MiscExtensions

open System
open System.Linq

type System.String with
    member self.StripIndent(): string =
        let lines = self.Replace("\r\n", "\n").Split('\n')
        let indent =
            lines
                .Skip(1)
                .Min(fun l -> l.TakeWhile(fun c -> c = ' ').Count())
        lines.[0] + Environment.NewLine +
            System.String.Join(Environment.NewLine,
                lines
                    .Skip(1)
                    .Select(fun l -> l.Substring(indent)))
